sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -51,6 +53,24 @@ fn verify_keys_packable<const VERIFY: bool>(
     Ok(())
 }
 
+fn verify_length_packable<const VERIFY: bool>(
+    map: &BTreeMapPrefix<
+        BoxedSlicePrefix<u8, MetadataFeatureKeyLength>,
+        BoxedSlicePrefix<u8, MetadataFeatureValueLength>,
+        MetadataFeatureLength,
+    >,
+) -> Result<(), Error> {
+    if VERIFY {
+        let len = map.packed_len();
+        if !MetadataFeature::LENGTH_RANGE
+            .contains(&u16::try_from(len).map_err(|e| Error::InvalidMetadataFeature(e.to_string()))?)
+        {
+            return Err(Error::InvalidMetadataFeature(len.to_string()));
```

Comment:
maybe add a bit more context to the string?, like `format!("Out of bounds byte Length: {}")`